odxtools/cli/compare.py

```diff
@@ -470,96 +457,82 @@ def compare_diagnostic_layers(self, dl1: DiagLayer,
                     # save information about changes in dictionary
 
                     # add new service (type: DiagService)
-                    service_dict["changed_name_of_service"][0].append(  # type: ignore[union-attr]
-                        service1)
+
+                    service_spec.changed_name_of_service[0].append(service1)
                     # add old service name (type: String)
-                    service_dict["changed_name_of_service"][1].append(  # type: ignore[union-attr]
-                        service2.short_name)
+                    service_spec.changed_name_of_service[1].append(service2.short_name)
```

Comment:
note to self: `changed_name_of_service` should become a list of dataclass objects featuring `.old_service_name` and `.new_service_name` attributes. (If you feel like it, I would appreciate if you implemented this here, but for present PR it is IMO not necessary...)